src/shims/foreign_items/posix/macos.rs

```diff
@@ -78,7 +78,8 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
                 let dtor = this.read_scalar(args[0])?.not_undef()?;
                 let dtor = this.memory.get_fn(dtor)?.as_instance()?;
                 let data = this.read_scalar(args[1])?.not_undef()?;
-                this.machine.tls.set_global_dtor(dtor, data)?;
+                let active_thread = this.get_active_thread()?;
+                this.machine.tls.set_global_dtor(active_thread, dtor, data)?;
```

Comment:
It's not really a "global" dtor any more when its per thread... so maybe rename this to "set_thread_dtor" or so?

Reply:
I have changed it to `set_thread_global_dtor` to make it still distinguishable from other dtors.